### PR TITLE
Add @camunda8/sdk test + housekeeping

### LINKED ISSUE #13: Failing test passes the CI
I am currently implementing a test with @camunda8/sdk. The test fails but the CI still reports a success: https://github.com/camunda/zeebe-connection-test/actions/runs/9174259442/job/25224729621
Perhaps it's due to `tee`.

### PR DESCRIPTION
**Pull request by barmac**
This PR adds @camunda8/sdk test, and also updates the CI to the latest supported versions of the actions and Node.
I also updated zeebe-node and zbctl deps.

Closes #13

### CODE REVIEW

**Comment by barmac** [comment]
I removed the Zeebe version update for now, and will provide it via a separate PR. With that change, the CI jobs wouldn't succeeed.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -66,12 +62,8 @@ jobs:
...
         ZEEBE_ADDRESS: test.example.com:443
       run: |
-        npm run test:secure | tee output.txt
-        results=$(grep "\"gatewayVersion\": \"${ZEEBE_VERSION}\"" -c output.txt)
-        if [[ "$results" != "2" ]]; then
-          echo "missing <gatewayVersion>"
-          exit 1
-        fi
+        set -o pipefail
+        npm run test:secure | tee output.txt ; echo $?
```
Comment (nikku) [inline .github/workflows/CI.yml]: I assume that `npm run test:**` execute with a non 0 exit code? Or how do we detect errors now?

Comment (barmac) [reply .github/workflows/CI.yml]: Before the change, `tee` would always return 0 exit code. So even if `npm run test:secure` failed, the job would succeed on GH Actions. With `set -o pipefail`, the pipe keeps non-zero exit code of the first command.

Comment (barmac) [reply .github/workflows/CI.yml]: False-positive job: https://github.com/camunda/zeebe-connection-test/actions/runs/9172668817/job/25219703471#step:10:131

Comment (barmac) [reply .github/workflows/CI.yml]: Failure with the new config: https://github.com/camunda/zeebe-connection-test/actions/runs/9177151880/job/25234078680#step:15:127

Comment (nikku) [reply .github/workflows/CI.yml]: > pipe keeps non-zero exit code of the first command.

This I understand. The previous test was not meant to propagate the exit code, but test the expected system behavior (gateway version reported). :slightly_smiling_face: 

What I'm saying is that we now rely on propagation of a `!= 0` exit code, which I'm OK with. I just want to make you aware that we change the test strategy.

Comment (barmac) [reply .github/workflows/CI.yml]: Let me check if we can make it work without change of the test strategy.

Comment (barmac) [reply .github/workflows/CI.yml]: No magic-number solution: https://github.com/camunda/zeebe-connection-test/pull/12/commits/806cf3a0aca9d79393e58b21b34ca32657b3454c

Comment (barmac) [reply .github/workflows/CI.yml]: So now build fails because updated `zbctl` prints double spaces between key and value 🙄

Comment (barmac) [reply .github/workflows/CI.yml]:
```json
> zeebe-tls-connection-test@0.1.0 test:zbctl
> zbctl status -o json

{
  "brokers":  [
    {
      "nodeId":  0,
      "host":  "0.0.0.0",
      "port":  26501,
      "partitions":  [
        {
          "partitionId":  1,
          "role":  "LEADER",
          "health":  "HEALTHY"
        }
      ],
      "version":  "8.5.1"
    }
  ],
  "clusterSize":  1,
  "partitionsCount":  1,
  "replicationFactor":  1,
  "gatewayVersion":  "8.5.1"
}
```

Comment (nikku) [reply .github/workflows/CI.yml]: Hahaha. Well, if we want to change our test strategy... :wink: 

Of course there is other ways to solve this, i.e. using [jq](https://jqlang.github.io/jq/) to parse the value returned.

Comment (barmac) [reply .github/workflows/CI.yml]: Fixed via https://github.com/camunda/zeebe-connection-test/pull/12/commits/aad4341b631e4e5e4c75d1b5af610a120ba2a51a

**Review by nikku — APPROVED** [review]
Looks great! :cake:

**Pull request merged**